oraclizeAPI_0.5.sol

```diff
@@ -77,760 +73,881 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query.value(price)(_timestamp, _datasource, _arg);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, string memory _arg, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource,_gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query_withGasLimit.value(price)(_timestamp, _datasource, _arg, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string memory _arg, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource, _gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+           return 0; // Unexpectedly high price
+        }
+        return oraclize.query_withGasLimit.value(price)(0, _datasource, _arg, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string memory _arg1, string memory _arg2) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2.value(price)(0, _datasource, _arg1, _arg2);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, string memory _arg1, string memory _arg2) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource);
+        if (price > 1 ether + tx.gasprice * 200000) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2.value(price)(_timestamp, _datasource, _arg1, _arg2);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, string memory _arg1, string memory _arg2, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource, _gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2_withGasLimit.value(price)(_timestamp, _datasource, _arg1, _arg2, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string memory _arg1, string memory _arg2, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        uint price = oraclize.getPrice(_datasource, _gasLimit);
+        if (price > 1 ether + tx.gasprice * _gasLimit) {
+            return 0; // Unexpectedly high price
+        }
+        return oraclize.query2_withGasLimit.value(price)(0, _datasource, _arg1, _arg2, _gasLimit);
+    }
+
+    function oraclize_query(string memory _datasource, string [] memory _argN) oraclizeAPI internal returns (bytes32 id) {

```

Comment:
can we avoid the extra whitespace between `string` and `[]`

Reply:
Sure can boss :+1: